pkg/lexer/lexer.go

```diff
@@ -0,0 +1,259 @@
+package lexer
+
+import (
+	"unicode"
+	"unicode/utf8"
+)
+
+type TokenType int
```

Comment:
I've reduced the API footprint to only have the types `TokenType` (+ constants), `Token`, `Lexer` and just one function `Lex()`. This makes the documentation straightforward (if we do end up writing a proper one) and doesn't interfere with all the fluff.